src/js/containers/covid19/awardSpendingAgency/AwardSpendingAgencyTableContainer.jsx

```diff
@@ -0,0 +1,222 @@
+            awardSpendingByAgencyRow.populate(item);
+
+            let rowChildren = [];
+            if (item.children && item.children.length > 0) {
+                rowChildren = item.children.map((childItem) => {
+                    const awardSpendingByAgencyChildRow = Object.create(BaseAwardSpendingByAgencyRow);
+                    awardSpendingByAgencyChildRow.populate(childItem);
+                    return awardSpendingByAgencyChildRow;
+                });
+            }
+
+            if (rowChildren && rowChildren.length > 0) {
+                Object.defineProperty(awardSpendingByAgencyRow, "children", {
+                    value: rowChildren
+                });
+            }
+
+            return awardSpendingByAgencyRow;
+        });
+        setResults(parsedData);
+    };
+
+    const fetchSpendingByCategoryCallback = useCallback(() => {
+        setLoading(true);
+
+        let params = {};
+
+        // if active tab is all, default to all award type codes
+        if (props.type === 'all') {
+            params = {
+                filter: {
+                    def_codes: defCodes.map((defc) => defc.code)
+                },
+                pagination: {
+                    limit: pageSize,
+                    page: currentPage,
+                    sort: awardSpendingAgencyTableColumnFieldMapping[sort],
+                    order
+                },
+                spending_type: 'award'
+            };
+        } else {
+            params = {
+                filter: {
+                    def_codes: defCodes.map((defc) => defc.code),
+                    award_type_codes: awardTypeGroups[props.type]
+                },
+                pagination: {
+                    limit: pageSize,
+                    page: currentPage,
+                    sort: awardSpendingAgencyTableColumnFieldMapping[sort],
+                    order
+                },
+                spending_type: 'award'
+            };
+        }
+
+        let faceValueOfLoansRequest;
+        if (props.type === 'loans') {
+            const faceValueOfLoansParams = {
+                filter: {
+                    def_codes: defCodes.map((defc) => defc.code)
+                },
+                pagination: {
+                    limit: pageSize,
+                    page: currentPage,
+                    sort: awardSpendingAgencyTableColumnFieldMapping[sort],
+                    order
+                },
+                spending_type: 'award'
+            };
+            faceValueOfLoansRequest = fetchFaceValueOfLoans(faceValueOfLoansParams);
+        }
+
+
+        if (faceValueOfLoansRequest) {
+            const request = fetchAwardSpendingByAgency(params);
+
+            // merge request arrays with face_value_of_loan data
+            Promise.all([request.promise, faceValueOfLoansRequest.promise]).then((res) => {
+                const result = res[0].data.results.filter((o1) =>
```

Comment:
Could we use more specific variable names here to improve readability? Something like `loanResult`/ `awardResult`/ `combinedResult`  instead of `o`/`o1`/`o2`